pkg/controllers/clusteroperator/clusteroperator_controller_test.go

```diff
@@ -68,62 +71,295 @@ var _ = Describe("ClusterOperator controller", func() {
+			Expect(err).ToNot(HaveOccurred(), "should be able to reconcile the cluster-api ClusterOperator without erroring")
+
+			By("Running checking the conditions are as expected")
+			Eventually(komega.Object(configv1resourcebuilder.ClusterOperator().WithName(controllers.ClusterOperatorName).Build())).
+				Should(HaveField("Status.Conditions", SatisfyAll(
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorAvailable)), HaveField("Status", Equal(configv1.ConditionTrue)),
+						HaveField("Message", Equal(fmt.Sprintf("Cluster CAPI Operator is available at %s", desiredOperatorReleaseVersion))))),
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorProgressing)), HaveField("Status", Equal(configv1.ConditionFalse)))),
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorDegraded)), HaveField("Status", Equal(configv1.ConditionFalse)))),
+					ContainElement(And(HaveField("Type", Equal(configv1.OperatorUpgradeable)), HaveField("Status", Equal(configv1.ConditionTrue)))),
+				)), "should match the expected ClusterOperator status conditions")
+		})
+
+		It("should aggregate the top level operator conditions with degraded=true when some controller conditions are degraded", func() {
+			By("Setting up the controllers conditions, one degraded")
+			startingCO := configv1resourcebuilder.ClusterOperator().WithName(controllers.ClusterOperatorName).Build()
+			patchBase := startingCO.DeepCopy()
+			startingCO.Status.Conditions = []configv1.ClusterOperatorStatusCondition{
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CoreClusterControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CoreClusterControllerDegradedCondition, Status: configv1.ConditionFalse, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.MachineSyncControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.MachineSyncControllerDegradedCondition, Status: configv1.ConditionFalse, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.KubeconfigControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.KubeconfigControllerDegradedCondition, Status: configv1.ConditionFalse, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CapiInstallerControllerAvailableCondition, Status: configv1.ConditionTrue, Reason: operatorstatus.ReasonAsExpected},
+				{LastTransitionTime: metav1.Now(), Type: operatorstatus.CapiInstallerControllerDegradedCondition, Status: configv1.ConditionTrue, Reason: "Error", Message: "reconciler error"},
```

Comment:
Maybe put a comment at the end of this line to indicate it's the one that's degraded, since these lines are all so similar. Otherwise it's commented 24 lines away.